auth/auth.go

```diff
@@ -183,6 +192,15 @@ func (auth *Authenticator) rebuildChannels(princ Principal) error {
 	}
 
 	base.Infof(base.KeyAccess, "Recomputed channels for %q: %s", base.UD(princ.Name()), base.UD(channels))
+
+	if channelsPerUserThreshold := auth.ChannelsPerUser(); channelsPerUserThreshold != nil {
+		channelCount := len(channels)
+		if uint32(channelCount) >= *channelsPerUserThreshold {
+			base.Warnf("User ID: %v channel count: %d exceeds %d for channels per user warning threshold",
```

Comment:
This is logged for roles too, which might be a valid thing to do, but the log message is confusing as it doesn't include the `role:` prefix for the principal name.

```
$ curl http://localhost:4985/db1/_role/bar
{"name":"bar","all_channels":["!","z"]}
```

```
2021-05-13T12:54:00.712+01:00 [WRN] User ID: bar channel count: 2 exceeds 2 for channels per user warning threshold -- auth.(*Authenticator).rebuildChannels() at auth.go:199
```

Reply:
We can get rid of by moving the logging to getPrincipal.